src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -382,7 +391,24 @@ class ResourceService(
     } else None
   }
 
-  private def fireGroupUpdateNotification(groupId: WorkbenchGroupIdentity, samRequestContext: SamRequestContext): Future[Unit] =
+  private[service] def validateDescendantPermissions(descendantPermissionsSet: Set[AccessPolicyDescendantPermissions]): IO[Seq[ErrorReport]] = {
+    val validationErrors = descendantPermissionsSet.toList.traverse { descendantPermissions =>
+      for {
+        maybeDescendantResourceType <- getResourceType(descendantPermissions.resourceType)
```

Comment:
`getResourceType` returns an `IO` which forces this function to return an `IO` which in turn forces functions up the stack to be or handle `IO`s